src/machine/machine_nrf52xxx.go

```diff
@@ -8,6 +8,48 @@ import (
 	"unsafe"
 )
 
+const (
+	SPI_MODE_CPHA0_CPOL0 SPIMode = iota
+	SPI_MODE_CPHA1_CPOL0
+	SPI_MODE_CPHA1_CPOL1
+	SPI_MODE_CPHA0_CPOL1
+
+	SPI_MODE_CPHA_FALLING_EDGE_CPOL_ACTIVE_LOW  = SPI_MODE_CPHA0_CPOL0
+	SPI_MODE_CPHA_RISING_EDGE_CPOL_ACTIVE_LOW   = SPI_MODE_CPHA1_CPOL0
+	SPI_MODE_CPHA_RISING_EDGE_CPOL_ACTIVE_HIGH  = SPI_MODE_CPHA1_CPOL1
+	SPI_MODE_CPHA_FALLING_EDGE_CPOL_ACTIVE_HIGH = SPI_MODE_CPHA0_CPOL1
+)
+
+// There are 3 SPI interfaces on the NRF528xx.
+var (
+	SPI0 = SPI{Bus: nrf.SPIM0}
+	SPI1 = SPI{Bus: nrf.SPIM1}
+	SPI2 = SPI{Bus: nrf.SPIM2}
+)
+
+type SPIMode uint8
```

Comment:
This is a breaking change. It must be discussed in an issue beforehand.

Reply:
Ok, will remove it from the config struct definition, so one can pass to the config but the type is not required.